api/admin/controller.py

```diff
@@ -462,6 +555,74 @@ def edit(self, identifier_type, identifier):
+                self._db.rollback()
+                return UNKNOWN_MEDIUM.detailed(
+                    _("Medium %(medium)s is not one of the known media.",
+                      medium=new_medium))
+            staff_edition.medium = new_medium
+            changed = True
+
+        new_language = flask.request.form.get("language")
+        if new_language:
+            new_language = LanguageCodes.string_to_alpha_3(new_language)
+            if not new_language:
+                self._db.rollback()
+                return UNKNOWN_LANGUAGE
+        if new_language != staff_edition.language:
+            staff_edition.language = new_language
+            changed = True
+
+        new_publisher = flask.request.form.get("publisher")
+        if new_publisher != staff_edition.publisher:
+            if staff_edition.publisher and not new_publisher:
+                new_publisher = NO_VALUE
+            staff_edition.publisher = unicode(new_publisher)
+            changed = True
+
+        new_imprint = flask.request.form.get("imprint")
+        if new_imprint != staff_edition.imprint:
+            if staff_edition.imprint and not new_imprint:
+                new_imprint = NO_VALUE
+            staff_edition.imprint = unicode(new_imprint)
+            changed = True
+
+        new_issued = flask.request.form.get("issued")
+        if new_issued:
+            try:
+                new_issued = datetime.strptime(new_issued, '%Y-%m-%d')
+            except ValueError:
+                self._db.rollback()
+                return INVALID_DATE_FORMAT
+        if new_issued != staff_edition.issued:
+            staff_edition.issued = new_issued
+            changed = True
+
+        # TODO: This lets library staff add a 1-5 rating, which is used in the
+        # quality calculation. However, this doesn't work well if there are any
+        # other measurements that contribute to the quality. The form will show
+        # the calculated quality rather than the staff rating, which will be
+        # confusing. It might also be useful to make it more clear how this
+        # relates to the quality threshold in the library settings.
+        new_rating = flask.request.form.get("rating")
+        if new_rating != None:
+            try:
+                new_rating = float(new_rating)
+            except ValueError:
+                self._db.rollback()
+                return INVALID_RATING
+            scale = Measurement.RATING_SCALES[DataSource.LIBRARY_STAFF]
+            if new_rating < scale[0] or new_rating > scale[1]:
+                self._db.rollback()
+                return INVALID_RATING.detailed(
+                    _("The rating must be a number between %(low)s and %(high)s.",
+                      low=scale[0], high=scale[1]))
+            if (new_rating - scale[0]) / (scale[1] - scale[0]) != work.quality:
+                primary_identifier.add_measurement(staff_data_source, Measurement.RATING, new_rating)
```

Comment:
You can also set `weight` really high here to swamp the effect of other ratings.